src/frontend/planner_test/tests/testdata/share.yaml

```diff
@@ -43,21 +43,19 @@
               ├─StreamExchange { dist: HashShard(id) }
               | └─StreamProject { exprs: [id, _row_id] }
               |   └─StreamFilter { predicate: (initial_bid = 1:Int32) }
-              |     └─StreamProject { exprs: [id, initial_bid, _row_id] }
-              |       └─StreamShare { id = 650 }
-              |         └─StreamProject { exprs: [id, initial_bid, _row_id] }
-              |           └─StreamFilter { predicate: ((initial_bid = 1:Int32) OR (initial_bid = 2:Int32)) }
-              |             └─StreamRowIdGen { row_id_index: 10 }
-              |               └─StreamSource { source: "auction", columns: ["id", "item_name", "description", "initial_bid", "reserve", "date_time", "expires", "seller", "category", "extra", "_row_id"] }
+              |     └─StreamShare { id = 658 }
+              |       └─StreamProject { exprs: [id, initial_bid, _row_id] }
+              |         └─StreamFilter { predicate: ((initial_bid = 1:Int32) OR (initial_bid = 2:Int32)) }
+              |           └─StreamRowIdGen { row_id_index: 10 }
+              |             └─StreamSource { source: "auction", columns: ["id", "item_name", "description", "initial_bid", "reserve", "date_time", "expires", "seller", "category", "extra", "_row_id"] }
```

Comment:
We can eliminate the project in this case since its parent isn't an exchange operator.